src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5866,20 +5883,50 @@ void SessionImpl::handleTorrentConflictAlert(const lt::torrent_conflict_alert *a
 
 void SessionImpl::processTrackerStatuses()
 {
+    if (m_updatedTrackerEntries.isEmpty())
+        return;
+
     for (auto it = m_updatedTrackerEntries.cbegin(); it != m_updatedTrackerEntries.cend(); ++it)
     {
-        auto torrent = static_cast<TorrentImpl *>(it.key());
-        const QSet<QString> &updatedTrackers = it.value();
+        invokeAsync([this, torrentHandle = it.key(), updatedTrackers = it.value()]() mutable
+        {
+            try
+            {
+                std::vector<lt::announce_entry> nativeTrackers = torrentHandle.trackers();
```

Comment:
> Why not move this inside the lambda below? You capture the handle anyway.

The lambda below is invoked in main thread whereas this line should be invoked in separate one.

Reply:
Maybe add a comment explaining the reason. To prevent future accidental relocation of this line.

Reply:
> To prevent future accidental relocation of this line.

But if you do this, it turns out that code is run in a separate thread that does nothing but call lambda in the main thread. Such a change would not seem suspicious to someone who would have planned to do this?
`invokeAsync()` is designed to execute asynchronous code (in a separate thread), so it seems superfluous to me to add additional comments about this fact every time it is used.

Reply:
Maybe add explanations on `invoke`/`invokeAsync` declarations?

Reply:
Hmm, I see your point. Probably there isn't a need for more comments.